lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -53,20 +53,19 @@ def objects
 
     def collection_ancestor? collections
       collections.each do |check|
-        return true if check.id == self.id
         return true if ancestor?(check)
       end
       false
     end
 
     def ancestor? collection
-      return true if collection.id == self.id
+      return true if collection == self
       return false if collection.child_collections.empty?
       current_collections = collection.child_collections
       next_batch = []
       while !current_collections.empty? do
         current_collections.each do |c|
-          return true if c.id == self.id
+          return true if c == self
```

Comment:
:+1: -- I did a quick check to satisfy myself that checking object equivalence works when you load a second copy of the collection (col2 = Hydra::PCDM::Collection.find(col1.pid)), and those two are equal, so that's all good.